test/e2e/storage/testsuites/volumelimits.go

```diff
@@ -0,0 +1,256 @@
+			ginkgo.Fail("Test driver does not provide dynamically created volumes")
+		}
+		claimSize := dDriver.GetClaimSize()
+
+		l.ns = f.Namespace
+		l.cs = f.ClientSet
+		l.config, l.testCleanup = driver.PrepareTest(f)
+		defer l.testCleanup()
+
+		l.resource = createGenericVolumeTestResource(driver, l.config, pattern)
+		defer l.resource.cleanupResource()
+		l.pvcs = append(l.pvcs, l.resource.pvc)
+
+		ginkgo.By("Creating single pod")
+		pod, err := framework.CreateSecPodWithNodeSelection(l.cs, l.ns.Name, []*v1.PersistentVolumeClaim{l.resource.pvc},
+			false, "", false, false, framework.SELinuxLabel,
+			nil, framework.NodeSelection{Name: l.config.ClientNodeName}, framework.PodStartTimeout)
+		framework.ExpectNoError(err, "failed to create the 1st pod")
+		l.pods = append(l.pods, pod)
+
+		ginkgo.By("Checking node limits")
+		nodeName := pod.Spec.NodeName
+		csiNode, err := l.cs.StorageV1beta1().CSINodes().Get(nodeName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		var csiDriver *storagev1beta1.CSINodeDriver
+		for _, c := range csiNode.Spec.Drivers {
+			if c.Name == driverInfo.Name {
+				csiDriver = &c
+				break
+			}
+		}
+		framework.ExpectNotEqual(csiDriver, nil, fmt.Sprintf("CSIDriver must contain driver %s", driverInfo.Name))
+		framework.ExpectNotEqual(csiDriver.Allocatable, nil, fmt.Sprintf("CSIDriver %s must contain Allocatable", driverInfo.Name))
+		framework.ExpectNotEqual(csiDriver.Allocatable.Count, nil, fmt.Sprintf("CSIDriver %s must contain Allocatable.Count", driverInfo.Name))
+		limit := int(*csiDriver.Allocatable.Count)
+
+		// Prepare cleanup
+		defer func() {
+			for _, pod := range l.pods {
+				err = l.cs.CoreV1().Pods(l.ns.Name).Delete(pod.Name, nil)
+				framework.ExpectNoError(err, "failed cleaning up pod")
+			}
+			for _, pvc := range l.pvcs {
+				err = l.cs.CoreV1().PersistentVolumeClaims(l.ns.Name).Delete(pvc.Name, nil)
+				framework.ExpectNoError(err, "failed cleaning up pvc")
+			}
+
+			// Wait for the PVs to be deleted. It includes also pod and PVC deletion because of PVC protection.
+			// We use PVs to make sure that the test does not leave orphan PVs when a CSI driver is destroyed
+			// just after the test ends.
+			err = wait.Poll(5*time.Second, testSlowMultiplier*framework.PVDeletingTimeout, func() (bool, error) {
+				existing := 0
+				for _, pvName := range l.pvNames.List() {
+					_, err = l.cs.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+					if err == nil {
+						existing++
+					} else {
+						if errors.IsNotFound(err) {
+							l.pvNames.Delete(pvName)
+						} else {
+							framework.ExpectNoError(err)
+						}
+					}
+				}
+				if existing > 0 {
+					e2elog.Logf("Waiting for %d PVs to be deleted", existing)
+					return false, nil
+				}
+				return true, nil
+			})
+			framework.ExpectNoError(err, "failed waiting for PVs to be deleted")
+		}()
+
+		// Create <limit> nr. of PVCs and pods. With one pod created above, we are 1 pod over the limit.
```

Comment:
I think we need more PVCs per pod otherwise we will hit 110 pods per node limit first

Reply:
Why is it so low???
Anyway, I was not able to even create 128 PVs:

```
I0717 13:28:50.828] STEP: Waiting for 129 PVCs Bound
I0717 13:29:00.642] Jul 17 13:29:00.642: INFO: 30/129 of PVCs are Bound
I0717 13:29:07.375] Jul 17 13:29:07.375: INFO: 37/129 of PVCs are Bound
I0717 13:29:15.760] Jul 17 13:29:15.759: INFO: 43/129 of PVCs are Bound
I0717 13:29:22.415] Jul 17 13:29:22.415: INFO: 52/129 of PVCs are Bound
I0717 13:29:30.550] Jul 17 13:29:30.549: INFO: 65/129 of PVCs are Bound
I0717 13:29:37.465] Jul 17 13:29:37.465: INFO: 72/129 of PVCs are Bound
I0717 13:29:45.482] Jul 17 13:29:45.482: INFO: 80/129 of PVCs are Bound
I0717 13:29:53.365] Jul 17 13:29:53.365: INFO: 87/129 of PVCs are Bound

...

I0717 13:59:11.115] Jul 17 13:59:11.115: INFO: 105/129 of PVCs are Bound
I0717 13:59:11.116] Jul 17 13:59:11.115: INFO: Unexpected error occurred: timed out waiting for the condition
```

Reply:
Are you sure it's not because pods couldn't be scheduled?  PD driver uses delayed binding